tests/protocol/test_wire.py

```diff
@@ -402,6 +402,39 @@ def test_fetch_manifest_fallback(self, *args):
                     self.assertEqual(patch_fetch.call_args_list[0][0][0], uri1.uri)
                     self.assertEqual(patch_fetch.call_args_list[1][0][0], host_uri)
 
+    def test_fetch_manifest_manifest_uri(self, *args):
+        uri1 = ExtHandlerVersionUri()
+        uri1.uri = 'ext_uri'
+        uris = DataContractList(ExtHandlerVersionUri)
+        uris.append(uri1)
+        host_uri = 'host_uri'
+        mock_host = HostPluginProtocol(host_uri,
+                                       'container_id',
+                                       'role_config')
+        client = WireProtocol(wireserver_url).client
+        manifest_return = "manifest.xml"
+
+        with patch.object(WireClient,
+                          "get_host_plugin",
+                          return_value=mock_host):
+            with patch.object(HostPluginProtocol,
+                              "get_artifact_request",
+                              return_value=[host_uri, {}]):
+
+                # First fetch works (direct download works)
+                with patch.object(WireClient, "fetch", return_value=manifest_return) as patch_fetch:
+                    self.assertEqual(client.fetch_manifest(uris), manifest_return)
```

Comment:
let's move the call to client.fetch out of the assert to make it clear who we are testing (AAA unit test pattern)